process/block/preprocess/scheduledTxsExecution_test.go

```diff
@@ -382,4 +391,609 @@ func TestScheduledTxsExecution_executeShouldWork(t *testing.T) {
+		&testscommon.MarshalizerMock{},
+		&mock.ShardCoordinatorStub{},
+	)
+
+	marshalledSCRs, err := scheduledTxsExec.getMarshalledScheduledRootHashSCRsGasAndFees(
+		scheduledRootHash, mapSCRs, gasAndFees,
+	)
+	assert.Nil(t, marshalledSCRs)
+	assert.Error(t, err)
+}
+
+func TestScheduledTxsExecution_getMarshalledScheduledRootHashSCRsGasAndFeesShouldWork(t *testing.T) {
+	t.Parallel()
+
+	scheduledRootHash := []byte("root hash")
+	mapSCRs := map[block.Type][]data.TransactionHandler{
+		0: {
+			&smartContractResult.SmartContractResult{
+				Nonce: 1,
+			},
+		},
+	}
+	gasAndFees := scheduled.GasAndFees{
+		AccumulatedFees: big.NewInt(100),
+	}
+
+	scheduledSCRs := &scheduled.ScheduledSCRs{
+		RootHash: scheduledRootHash,
+		Scrs: map[int32]scheduled.SmartContractResults{
+			0: {
+				TxHandlers: []*smartContractResult.SmartContractResult{
+					{
+						Nonce: 1,
+					},
+				},
+			},
+		},
+		GasAndFees: &gasAndFees,
+	}
+	expectedScheduledSCRs, _ := json.Marshal(scheduledSCRs)
+
+	scheduledTxsExec, _ := NewScheduledTxsExecution(
+		&testscommon.TxProcessorMock{},
+		&mock.TransactionCoordinatorMock{},
+		&genericMocks.StorerMock{},
+		&testscommon.MarshalizerMock{},
+		&mock.ShardCoordinatorStub{},
+	)
+
+	marshalledSCRs, err := scheduledTxsExec.getMarshalledScheduledRootHashSCRsGasAndFees(
+		scheduledRootHash, mapSCRs, gasAndFees,
+	)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedScheduledSCRs, marshalledSCRs)
+}
+
+func TestScheduledTxsExecution_RollBackToBlockShouldFail(t *testing.T) {
+	t.Parallel()
+
+	rootHash := []byte("root hash")
+
+	expectedErr := errors.New("local err")
+	scheduledTxsExec, _ := NewScheduledTxsExecution(
+		&testscommon.TxProcessorMock{},
+		&mock.TransactionCoordinatorMock{},
+		&storageMocks.StorerStub{
+			GetCalled: func(_ []byte) ([]byte, error) {
+				return nil, expectedErr
+			},
+		},
+		&marshal.GogoProtoMarshalizer{},
+		&mock.ShardCoordinatorStub{},
+	)
+
+	err := scheduledTxsExec.RollBackToBlock(rootHash)
+	assert.Equal(t, expectedErr, err)
+}
+
+func TestScheduledTxsExecution_RollBackToBlockShouldWork(t *testing.T) {
+	t.Parallel()
+
+	headerHash := []byte("root hash")
+
+	scheduledSCRs := &scheduled.ScheduledSCRs{
+		RootHash: headerHash,
+		Scrs: map[int32]scheduled.SmartContractResults{
+			0: {
+				TxHandlers: []*smartContractResult.SmartContractResult{},
+			},
+		},
+		GasAndFees: &scheduled.GasAndFees{
+			AccumulatedFees: big.NewInt(100),
+		}}
+	marshalledSCRsSavedData, _ := json.Marshal(scheduledSCRs)
+
+	scheduledTxsExec, _ := NewScheduledTxsExecution(
+		&testscommon.TxProcessorMock{},
+		&mock.TransactionCoordinatorMock{},
+		&storageMocks.StorerStub{
+			GetCalled: func(_ []byte) ([]byte, error) {
+				return marshalledSCRsSavedData, nil
+			},
+		},
+		&testscommon.MarshalizerMock{},
+		&mock.ShardCoordinatorStub{},
+	)
+
+	err := scheduledTxsExec.RollBackToBlock(headerHash)
+	assert.Nil(t, err)
+
+	hash, err := scheduledTxsExec.GetScheduledRootHashForHeader(headerHash)
```

Comment:
can you do a getScheduledRootHasGasAndFees, and check all expected values?

Reply:
done